OCPBUGS#16299: Add AWS Load Balancer Operator considerations (#62497)
openshift/openshift-docs: modules/nw-aws-load-balancer-operator-considerations.adoc

// Module included in the following assemblies:
// * networking/aws_load_balancer_operator/understanding-aws-load-balancer-operator.adoc

:_content-type: REFERENCE
[id="nw-aws-load-balancer-operator-considerations_{context}"]
= AWS Load Balancer Operator considerations

Review the following limitations before installing and using the AWS Load Balancer Operator.

> * The IP traffic mode only works on AWS Elastic Kubernetes Service (EKS). The AWS Load Balancer Operator disables the IP traffic mode for the AWS Load Balancer Controller and hence, the AWS Load Balancer Controller cannot use the pod readiness gate.

Reviewer: ```suggestion
* The IP traffic mode only works on AWS Elastic Kubernetes Service (EKS). The AWS Load Balancer Operator disables the IP traffic mode for the AWS Load Balancer Controller and hence, the AWS Load Balancer Controller cannot use the pod readiness gate either.
```
Just wanted to highlight that 2 things are not working: ip mode and the readiness gate. To me it sounds a little like we disable the ip mode because we don't want to use the pod readiness. Or it's only me?

> * After installing the AWS Load Balancer Operator, by default, controller command-line flags such as `--disable-ingress-class-annotation` and `--disable-ingress-group-name-annotation` are added to the AWS Load Balancer Controller. Therefore, the AWS Load Balancer Operator does not allow using the `kubernetes.io/ingress.class` and `alb.ingress.kubernetes.io/group.name` annotations in the `AWSLoadBalancerController` custom resource (CR).

Reviewer: ```suggestion
* The AWS Load Balancer Operator  adds command-line flags `--disable-ingress-class-annotation` and `--disable-ingress-group-name-annotation` to the AWS Load Balancer Controller. Therefore, the AWS Load Balancer Operator does not allow using the `kubernetes.io/ingress.class` and `alb.ingress.kubernetes.io/group.name` annotations in the `Ingress` resource.
```
ALBO adds those flags all the time, there is no way to customize it, so I think that `by default` may mislead a little.
The annotations will be applied to `Ingress` resource, sorry, I didn't see it the first time.

// Module included in the following assemblies:
// * networking/aws_load_balancer_operator/understanding-aws-load-balancer-operator.adoc

:_content-type: REFERENCE
[id="nw-aws-load-balancer-operator-considerations_{context}"]
= AWS Load Balancer Operator considerations

> Review the following limitations before installing and using the AWS Load Balancer Operator (ALBO).

Reviewer: The naming `ALBO` is quite commonly used inside Red Hat engineering community but I'm not sure whether it's an appropriate acronym for the OCP docs taking into account that the operator seems to be always called by its full name.

> * After installing the ALBO, by default, controller command-line flags such as link:https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/deploy/configurations/#disable-ingress-class-annotation[`--disable-ingress-class-annotation`] and link:https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/deploy/configurations/#disable-ingress-group-name-annotation[`--disable-ingress-group-name-annotation`] are added to the AWS Load Balancer Controller. Therefore, the ALBO does not allow using the `kubernetes.io/ingress.class` and `kubernetes.io/ingress.class` annotations in the `AWSLoadBalancerController` custom resource (CR).

Reviewer: Again, I think that we can mention the flags without the link to the upstream.

```suggestion
* After installing the ALBO, by default, controller command-line flags such as link:https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/deploy/configurations/#disable-ingress-class-annotation[`--disable-ingress-class-annotation`] and link:https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/deploy/configurations/#disable-ingress-group-name-annotation[`--disable-ingress-group-name-annotation`] are added to the AWS Load Balancer Controller. Therefore, the ALBO does not allow using the `kubernetes.io/ingress.class` and `alb.ingress.kubernetes.io/group.name` annotations in the `AWSLoadBalancerController` custom resource (CR).
```

> * The IP target mode only works on AWS Elastic Kubernetes Service (EKS). The ALBO disables the link:https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/how-it-works/#ip-mode[IP target mode] for the AWS Load Balancer Controller and hence, the AWS Load Balancer Controller does not use the link:https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/deploy/pod_readiness_gate/[pod readiness gate].

Reviewer: Let's put this line first because it's more important.

I would like to abstain from linking the upstream documentation. We can just say `IP traffic mode` (`traffic` is more commonly used than `target`) and `pod readiness gate`. The reader should understand the context. But up to you to clarify the point about the linking to the upstream docs.

> * The IP target mode only works on AWS Elastic Kubernetes Service (EKS). The ALBO disables the link:https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/how-it-works/#ip-mode[IP target mode] for the AWS Load Balancer Controller and hence, the AWS Load Balancer Controller does not use the link:https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/deploy/pod_readiness_gate/[pod readiness gate].

Reviewer: ```suggestion
* The IP target mode only works on AWS Elastic Kubernetes Service (EKS). The ALBO disables the link:https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/how-it-works/#ip-mode[IP target mode] for the AWS Load Balancer Controller and hence, the AWS Load Balancer Controller cannot make use of the link:https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/deploy/pod_readiness_gate/[pod readiness gate].
```

[NOTE]
====

> The AWS Load Balancer Operator version 1.0.0 supports the AWS Load Balancer Controller version 2.4.

Reviewer: Does it belong to the release notes chapter more than here? WDYT?
